File: examples/greeting-service/src/test/java/greeting/example/GreetingClient.java
package greeting.example;


import io.micronaut.http.client.annotation.Client;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Nullable;

@Client("/")
public interface GreetingClient {

    @GetMapping("/greeting{?name}")
    Greeting greet(@Nullable String name);

    @PostMapping("/greeting")
    Greeting greetByPost(@RequestBody Greeting greeting);

    @DeleteMapping("/greeting")
    HttpStatus deletePost();

    @GetMapping("/nested/greeting{?name}")
    Greeting nestedGreet(@Nullable String name);

    @GetMapping("/greeting-status{?name}")
    ResponseEntity<Greeting> greetWithStatus(@Nullable String name);
}

File: examples/greeting-service-kotlin/src/test/kotlin/greeting/example/GreetingClient.kt
package greeting.example


import io.micronaut.http.HttpStatus
import io.micronaut.http.client.annotation.Client
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import javax.annotation.Nullable

@Client("/")
interface GreetingClient {

    @GetMapping("/greeting{?name}")
    fun greet(@Nullable name : String? ) : Greeting

    @PostMapping("/greeting")
    fun greetByPost(@RequestBody greeting : Greeting) : Greeting

    @DeleteMapping("/greeting")
    fun deletePost() : HttpStatus

    @GetMapping("/nested/greeting{?name}")
    fun nestedGreet(@Nullable name : String?) : Greeting

    @GetMapping("/greeting-status{?name}")
    fun greetWithStatus(@Nullable name : String?) : ResponseEntity<Greeting>
}
